src/typescript/sdk/tests/e2e/arena/general/arena.test.ts

```diff
@@ -497,3 +553,179 @@ describe("ensures leaderboard history is working", () => {
+      melee.market1.symbolEmojis,
+      melee.market2.symbolEmojis,
+      "symbol1"
+    );
+    melee = await fetchArenaMeleeView(res.arena.event.meleeID).then(fetchMeleeEmojiData);
+    await waitForProcessor(res);
+
+    return true;
+  }, 30000);
+
+  it("verifies that arena info data is correct 1", async () => {
+    const account1 = getFundedAccount("420");
+    const account2 = getFundedAccount("421");
+    const account3 = getFundedAccount("422");
+
+    let volume = 0n;
+
+    const account1EnterRes = await enterHelper(account1, "symbol1");
+    const account2EnterRes = await enterHelper(account2, "symbol1");
+    const account3EnterRes = await enterHelper(account3, "symbol1");
+
+    volume += account1EnterRes.events.arenaEnterEvents[0].quoteVolume;
+    volume += account2EnterRes.events.arenaEnterEvents[0].quoteVolume;
+    volume += account3EnterRes.events.arenaEnterEvents[0].quoteVolume;
+
+    await emojicoin.arena.exit(account1, melee.market1.symbolEmojis, melee.market2.symbolEmojis);
+    const swapRes = await emojicoin.arena.swap(
+      account2,
+      melee.market1.symbolEmojis,
+      melee.market2.symbolEmojis
+    );
+    volume += swapRes.events.arenaSwapEvents[0].quoteVolume;
+    const res = await emojicoin.arena.exit(
+      account2,
+      melee.market1.symbolEmojis,
+      melee.market2.symbolEmojis
+    );
+    await waitForProcessor(res);
+
+    const arenaInfo = await fetchArenaInfo({});
+
+    expect(arenaInfo).toBeTruthy();
+    expect(arenaInfo!.meleeID).toEqual(melee.view.meleeID);
+    expect(arenaInfo!.duration).toEqual(melee.view.duration);
+    expect(arenaInfo!.startTime).toEqual(melee.view.startTime);
+    expect(arenaInfo!.maxMatchAmount).toEqual(melee.view.maxMatchAmount);
+    expect(arenaInfo!.maxMatchPercentage).toEqual(melee.view.maxMatchPercentage);
+    expect(arenaInfo!.volume).toEqual(volume);
+    expect(arenaInfo!.emojicoin0Locked).toEqual(
+      account3EnterRes.events.arenaEnterEvents[0].emojicoin0Proceeds
+    );
+    expect(arenaInfo!.emojicoin1Locked).toEqual(0n);
+  }, 30000);
+
+  it("verifies that arena info data is correct 2", async () => {
+    let volume = 0n;
+    let emojicoin0Locked = 0n;
+    let emojicoin1Locked = 0n;
+
+    const accounts = [
+      getFundedAccount("420"),
+      getFundedAccount("421"),
+      getFundedAccount("422"),
+      getFundedAccount("423"),
+      getFundedAccount("424"),
+      getFundedAccount("425"),
+      getFundedAccount("426"),
+      getFundedAccount("427"),
+    ];
+
+    for (const account of accounts.map((a, i) => [a, i] as [Ed25519Account, number])) {
+      const enterRes = await enterHelper(
+        account[0],
+        `symbol${(account[1] % 2) + 1}` as "symbol1" | "symbol2"
+      );
+      volume += enterRes.events.arenaEnterEvents[0].quoteVolume;
+      emojicoin0Locked += enterRes.events.arenaEnterEvents[0].emojicoin0Proceeds;
+      emojicoin1Locked += enterRes.events.arenaEnterEvents[0].emojicoin1Proceeds;
+    }
```

Comment:
```suggestion
    for (const [account, index] accounts.map((a, i) => [a, i] as [Ed25519Account, number])) {
      const enterRes = await enterHelper(
        account,
        // Alternate each account between entering with "symbol1" and "symbol2".
        `symbol${(index % 2) + 1}` as "symbol1" | "symbol2"
      );
      volume += enterRes.events.arenaEnterEvents[0].quoteVolume;
      emojicoin0Locked += enterRes.events.arenaEnterEvents[0].emojicoin0Proceeds;
      emojicoin1Locked += enterRes.events.arenaEnterEvents[0].emojicoin1Proceeds;
    }
```

Confusing what this is doing at first, you can unpack the values in the iterator like this to make it a little more clear